src/test/fs_tests.cpp

```diff
@@ -118,4 +118,37 @@ BOOST_AUTO_TEST_CASE(fsbridge_fstream)
     }
 }
 
-BOOST_AUTO_TEST_SUITE_END()
\ No newline at end of file
+BOOST_AUTO_TEST_CASE(rename)
+{
+    const fs::path tmpfolder{m_args.GetDataDirBase()};
+
+    const fs::path path1{GetUniquePath(tmpfolder)};
+    const fs::path path2{GetUniquePath(tmpfolder)};
+
+    const std::string path1_contents{"1111"};
+    const std::string path2_contents{"2222"};
+
+    {
+        std::ofstream file{path1};
+        file << path1_contents;
+    }
+
+    {
+        std::ofstream file{path2};
+        file << path2_contents;
+    }
+
+    // Rename path1 -> path2.
+    BOOST_CHECK(RenameOver(path1, path2));
+
+    BOOST_CHECK(!fs::exists(path1));
+
+    {
+        std::ifstream file{path2};
+        std::string contents;
+        file >> contents;
+        BOOST_CHECK_EQUAL(contents, path1_contents);
+    }
```

Comment:
It is a good practice to clean up after the test, even if it is a temporary folder. I forgot that, but woke up during the night with the thought "file cleanup was forgotten!".

```suggestion
    }
    fs::remove(path2);
```

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/24331#issuecomment-1038872961).